sys/stdio_uart/stdio_uart.c

```diff
@@ -43,10 +43,8 @@
 #define ENABLE_DEBUG 0
 #include "debug.h"
 
-#ifdef MODULE_STDIO_UART_RX
 static uint8_t _rx_buf_mem[STDIO_UART_RX_BUFSIZE];
 isrpipe_t stdio_uart_isrpipe = ISRPIPE_INIT(_rx_buf_mem);
-#endif
```

Comment:
won't this lead to `stdio_uart_isrpipe = ISRPIPE_INIT(_rx_buf_mem);` always require memory (global mem (it has no static)) even if the user does not want `RX` for his `STDIO_UART` (i am not sure about the usecase but for some reason that is a separate pseudo module)

Reply:
No, the compiler flag `-fdata-sections` forces every variable in its own linker section, so that it can be garbage collected at link time if there is no reference to it.